File: 2017/01.py

import math, copy, re


def part_1(data):
	L = len(data)
	suma = 0

	for i in range(L):
		if data[i] == data[(i + 1) % L]:
			suma += data[i]
	print(suma)
	return

def part_2(data):
	L = len(data)
	half = L // 2
	suma = 0

	for i in range(L):
		if data[i] == data[(i + half) % L]:
			suma += data[i]
	print(suma)

	return 


if __name__ == '__main__':
	with open('1_input') as f:
		data = f.read()
		# data = data.split()
		data = list(map(int, list(data)))


	part_1(data)
	part_2(data)
	

File: 2015/01.js

const fs = require('fs');
const { exec } = require("child_process");
const { bin, float, hex, int, num, oct } = require("../lib");
const { range, drange, trange, iter, diter, titer } = require("../lib");
const { copy,	entries, in_, inBB, keys, sort, values } = require("../lib");
const { Counter, DD, empty, FastQueue, PriorityQueue, Queue, set, Stack } = require("../lib");
const { ord, chr, count, debug, disp, crt, gcd, lcm, modPow, mod, prod, prodBig, randint, sum, sumBig, transpose } = require("../lib");
const { digits, ints,	floats,	singles,	words,	lines,	table,	groups,	getGroups,	groupsWith,	parse,	parseLine } = require ('../lib');
const { min, max, random, abs, ceil, floor, log, log10, log2, round, sign, sin, cos, tan, asin, acos, atan, atan2, sqrt, PI } = Math;
const { isSuperset, or, and, xor, sub } = set;
const { getExecStr } = require("../lib/post");
// const { parsePacket, ops, HEX, ALU, VARS, resetVars } = require('./lib_2021'); // specific to AOC 2021

function part1(data) {

	let res = 0;
	data = singles(data);
	// debug(data)
	iter(data, b => (res += (b === '(' ? 1 : -1)) && false)

	debug(res);
	exec(`echo ${res} | xclip -sel clip -rmlastnl`);
	console.log("END OF PART1");
	return;
}

function part2(data) {

	let res = 0;
	iter(data, (b, i) => {
		res += b === '(' ? 1 : -1;
		if (res < 0) {
			debug(i + 1);
			return true;
		}
	})

	// debug(res);
	// exec(`echo ${res} | xclip -sel clip -rmlastnl`);
	console.log("END OF PART2");
	return;
}

function main() {
	let data = fs.readFileSync("01_input").toString("utf-8");

	part1(data);
	part2(data);
	process.exit(0);
}

main();
